plat/common/x86/syscall.S

```diff
@@ -65,54 +67,110 @@ ENTRY(_ukplat_syscall)
 	 * in the Red Zone.
 	 */
 	movq	%gs:LCPU_AUXSP_OFFSET, %rsp
+
 	/* Describing the rsp relative to GS would make it necessary to emit
 	 * raw CFI. Instead of doing so, mark rsp as undefined temporarily
 	 */
 	.cfi_undefined rsp
 
+	subq	$(UKARCH_AUXSPCB_SIZE), %rsp
+	movq	UKARCH_AUXSPCB_OFFSETOF_CURR_FP(%rsp), %rsp
+
+	/* We subtract 8 less bytes here so that we have room for the pushed
+	 * auxsp.
+	 * Current stack pointer now points to the current frame pointer
+	 * within the auxiliary stack. Subtract UKARCH_EXECENV_END_ALIGN
+	 * to make room for the 8-byte auxsp pointer, since the layout is
+	 * struct uk_syscall_ctx {
+	 *	struct ukarch_execenv execenv;
+	 *	__uptr auxsp;  <-- make room for these
+	 *     ..... leftover room because of alignment subtraction ....
```

Comment:
```diff
-..... leftover room because of alignment subtraction ....
+... padding to ensure alignment ...
```

Reply:
Hmm, I intentionally said "leftover" because it is an unfortunate consequence of having to create a 64 byte slot only to store an 8 byte value just because of alignment constraints.  Padding sounds like it's something that we intentionally add? I don't know :thinking:

Reply:
Okay, understand. How about: "space left unused"... somehow trying to avoid "leftover" because it sounds like a gift for somebody - not sure if only I am thinking that way 😄